openmc/material.py

```diff
@@ -1181,22 +1172,38 @@ def export_to_xml(self, path='materials.xml'):
             Path to file to write. Defaults to 'materials.xml'.
 
         """
-
-        root_element = ET.Element("materials")
-        self._create_cross_sections_subelement(root_element)
-        self._create_material_subelements(root_element)
-
-        # Clean the indentation in the file to be user-readable
-        clean_indentation(root_element)
-
         # Check if path is a directory
         p = Path(path)
         if p.is_dir():
             p /= 'materials.xml'
 
-        # Write the XML Tree to the materials.xml file
-        tree = ET.ElementTree(root_element)
-        tree.write(str(p), xml_declaration=True, encoding='utf-8')
+        # Open the file in write mode.
+        with open(str(p), 'w', encoding='utf-8',
+                  errors='xmlcharrefreplace') as fh:
+
+            # Write the header and the opening tag for the root element.
+            fh.write("<?xml version='1.0' encoding='utf-8'?>\n")
+            fh.write('<materials>\n')
+
+            # Write the <cross_sections> element.
+            if self._cross_sections is not None:
+                element = ET.Element('cross_sections')
+                element.text = str(self._cross_sections)
+                clean_indentation(element, level=1)
+                element.tail = element.tail.strip(' ')
+                fh.write('  ')
+                ET.ElementTree(element).write(fh, encoding='unicode')
```

Comment:
Would this be better?
```suggestion
                fh.write(ET.tostring(fh, encoding='unicode'))
```

Reply:
`ET.tostring` actually just calls exactly this `ET.ElementTree(element).write` under the hood (but with an `io.StringIO`) so I figured I would just cut-out the middle man.  I doubt it makes a big difference.

Reply:
Oh, haha. Fine by me then :+1: